src/mem/alloc.h

```diff
@@ -948,7 +823,7 @@ namespace snmalloc
         return;
 
       stats().remote_post();
-      remote.post(&large_allocator, get_trunc_id());
+      remote.post<Allocator>(this, get_trunc_id());
```

Comment:
The problem is the type of `large_allocator` also depends on the template.  So I think what you are proposing could be done with two template parameters.  But then I worry that it would be harder to follow.  But I didn't like making it a friend.